src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -301,11 +303,26 @@ private string BuildDocumentId(string id, long newEtag, out bool knownNewId)
                         i += 19;
                         valuePtr[i] = '-';
 
-                        Format.Backwards.WriteNumber(valuePtr + i - 1, (ulong)newEtag);
+                        while (true)
+                        {
+                            Format.Backwards.WriteNumber(valuePtr + i - 1, (ulong)newEtag);
+
+                            using (DocumentIdWorker.GetSliceFromId(context, value, out var lowerId))
```

Comment:
We don't want it here.
This is a perf critical section of the code.
The whole point of _knownNewId_ is that we can skipping checking the storage.
That can give us some significant perf gains.

This check needs to happen during smuggler only (in the smuggler code).
We already have a warning about this issue in the docs, avoid using this style there.
However, we need to handle the case of the user just wanting to import data from another node and hitting this issue.